### ISSUE TITLE
Try to rewrite VDSR code by using tensorlayer, pls help!

### ISSUE THREAD

**Issue by xx4966946** [tensorlayer/tensorlayer#299]
I tried to rewrite VDSR code by using tensorlayer but it does not work. The result is weird. Please help. I really appreciate it.

The code I try to rewrite is from 'https://github.com/Jongchan/tensorflow-vdsr/blob/master/MODEL.py'

My code is:

`
import tensorflow as tf
import tensorlayer as tl
from tensorlayer.layers import *
import numpy as np

def VDSR(x, is_train=False, reuse=False):

    w_init = tf.random_normal_initializer(stddev=np.sqrt(2.0/9))
    w_init1 = tf.random_normal_initializer(stddev=np.sqrt(2.0/9/64))
    b_init = tf.constant_initializer(value=0.0)

    with tf.variable_scope("VDSR", reuse=reuse):
        tl.layers.set_name_reuse(reuse)
        inputs = InputLayer(x, name='inputs')
        conv1 = Conv2d(inputs, 64, (3, 3), act=tf.nn.relu, W_init=w_init, b_init=b_init, name='conv1_1')
        n = conv1

        for i in range(18):
            conv2 = Conv2d(n, 64, (3, 3), act=tf.nn.relu, W_init=w_init1, b_init=b_init, name='conv2_%s' % i)
            n = conv2

        conc3 = Conv2d(n, 1, (3, 3), act=None,  W_init=w_init1, b_init=b_init, name='conv3_1')

        output = ElementwiseLayer([conv3, inputs], tf.add, 'add1')
    return output

`

**Comment by zsdonghao** [comment]
Hi, i just have a quick check, it seems your code is correct, what is the error?

this code may look simpler.
```python
def model(x, reuse):
    with tf.variable_scope("VDSR", reuse=reuse):
        tl.layers.set_name_reuse(reuse)
        ni = InputLayer(x, name='inputs')
        n = Conv2d(ni, 64, (3, 3), act=tf.nn.relu, W_init=w_init, b_init=b_init, name='conv1_1')
        for i in range(18):
            n = Conv2d(n, 64, (3, 3), act=tf.nn.relu, W_init=w_init1, b_init=b_init, name='conv2_%s' % i)
        n = Conv2d(n, 1, (3, 3), act=None,  W_init=w_init1, b_init=b_init, name='conv3_1')
        n = ElementwiseLayer([n, ni], tf.add, 'add1')
    return n
```

**Comment by xx4966946** [comment]
This model is a super resolution model. I am using it to do something like denoise image. The original VDSR model can make the input smooth and reduce noise, but when I try to use tensorlayer to reproduce, the result is super blur (even worse than the input noisy image), cannot figure out why (I set all paras same, also learning rate).

**Comment by zsdonghao** [comment]
@xx4966946 Could you check whether your code update all parameters of the network?

You can feed all parameters with the name of "VDSR" to the optimiser as follow:

```python
train_params = tl.layers.get_variables_with_name('VDSR', train_only=True, printable=False)
train_op = tf.train.AdamOptimizer(learning_rate=0.0001).minimize(cost, var_list=train_params)
```

**Comment by xx4966946** [comment]
I made a really stupid mistake. I feed wrong ROI between training image and ground truth image.

Everything is going well right now. Thanks for your help!
